src/backends/pyplot.jl

```diff
@@ -937,13 +937,21 @@ function _before_layout_calcs(plt::Plot{PyPlotBackend})
                 kw[:boundaries] = vcat(0, kw[:values] + 0.5)
             elseif any(colorbar_series[attr] !== nothing for attr in (:line_z, :fill_z, :marker_z))
                 cmin, cmax = get_clims(sp)
-                norm = pycolors."Normalize"(vmin = cmin, vmax = cmax)
+                norm = if sp[:colorbar_scale] === :identity
+                    pycolors."Normalize"(vmin = cmin, vmax = cmax)
+                elseif sp[:colorbar_scale] === :log10
+                    pycolors."LogNorm"(vmin = cmin, vmax= cmax)
+                else
+                    @warn("Other than identity, log10 scales are not supported by matplotlib, defaulting to identity")
+                    pycolors."Normalize"(vmin = cmin, vmax = cmax)
+                end
+
                 f = if colorbar_series[:line_z] !== nothing
-                    py_linecolormap
+                    x -> py_linecolormap(x, scale=sp[:colorbar_scale])
                 elseif colorbar_series[:fill_z] !== nothing
-                    py_fillcolormap
+                    x -> py_fillcolormap(x, scale=sp[:colorbar_scale])
                 else
-                    py_markercolormap
+                    x -> py_markercolormap(x, scale=sp[:colorbar_scale])
```

Comment:
I thought the marker color would be calculated here, but it presumably gets calculated somewhere early in the pipeline and I do not know where.